plugin/core/registry.py

```diff
@@ -226,6 +228,37 @@ def run_async() -> None:
         sublime.set_timeout_async(run_async)
 
 
+class LspCheckApplicableCommand(sublime_plugin.TextCommand):
+
+    def run(self, edit: sublime.Edit, session_name: str) -> None:
+        sublime.set_timeout_async(lambda: self._run_async(session_name))
+
+    def _run_async(self, session_name: str) -> None:
+        if wm := windows.lookup(self.view.window()):
+            config = wm.get_config_manager().get_config(session_name)
+            if not config:
+                debug(f'Configuration with name {session_name} does not exist')
+                return
+            listener = windows.listener_for_view(self.view)
+            if not listener:
+                debug(f'No listener for view {self.view}')
+                return
+            scheme, _ = parse_uri(uri_from_view(self.view))
+            is_applicable = config.match_view(self.view, scheme)
+            if session := wm.get_session(session_name, self.view.file_name() or ''):
+                session_view = session.session_view_for_view_async(self.view)
+                if is_applicable and not session_view:
+                    listener.on_session_initialized_async(session)
+                elif not is_applicable and session_view:
+                    session.shutdown_session_view_async(session_view)
+            elif is_applicable:
+                wm.start_async(config, self.view)
+                if wm._new_session:
+                    wm._sessions.add(wm._new_session)
+                    listener.on_session_initialized_async(wm._new_session)
+                    wm._new_session = None
```

Comment:
This part is pretty ugly, but the public `WindowManager.start_async` method is not actually sufficient to start a new session. That method sets an internal variable at https://github.com/sublimelsp/LSP/blob/d5aefde238d51ed3d6b3562b14380a6b5e6fe784/plugin/core/windows.py#L280 and it expects some more code to be run after that, in order to properly handle the start of a new session.

I won't touch the recursive https://github.com/sublimelsp/LSP/blob/d5aefde238d51ed3d6b3562b14380a6b5e6fe784/plugin/core/windows.py#L154 method because it is a riddle to me how that works...

Starting a new session when triggered from the `lsp_check_applicable` TextCommand seems to work correctly now though.